.travis.yml

```diff
@@ -146,6 +147,28 @@ script:
        travis_terminate 0; 
     fi;
 
+  # For the trunk use TRAVIS_COMMIT_RANGE but "Auto cancel branch builds" has to be turned off at https://travis-ci.org/percona/percona-server/settings
+  # For pull requests and feature branches replace TRAVIS_COMMIT_RANGE with the range from the root to the tip of the branch
+  - if [[ "$TRAVIS_EVENT_TYPE" == "pull_request" ]] || [[ "$TRAVIS_BRANCH" != "$PARENT_BRANCH" ]]; then
```

Comment:
May be detecting development branches similarly to what we did in `if:` clauses in the matrix
`TRAVIS_REPO_SLUG != percona/percona-server`
There is a chance that somebody will work directly with his main branches (5.5/5.6/5.7) in his fork.